### PR TITLE
Fix mac bundle

### PR DESCRIPTION
**Pull request by omichel**
This PR aims at fixing the Webots.app bundle on macOS which is displayed as "damaged" when it contains a universal binary for webots (arm64 & x86_64).

1. Universal apps needs to be signed, therefore I added the `codesign` step for every binary.
2. In order to sign a macOS bundle, the bundle has to respect the Apple bundle folder hierarchy which was wrong, so I fixed it as well.
3. I upgraded to Python 3.10.5.
4. While testing I also found and fixed a bug preventing the detection of the most recent installed version of MATLAB.

### CODE REVIEW

**Review by ygoumaz — COMMENTED** [review]
I have some minor suggestions/questions on the code. I tested the compilation and distribution on Linux and it works fine. I still need to test on MacOS.

File: scripts/packaging/mac_distro.py [hunk 1]
```diff
@@ -159,19 +186,31 @@ def create_webots_bundle(self):
 
         print('Done.\n')
 
-    def make_dir(self, directory):
+    def make_dir(self, directory=''):
```
Comment (stefaniapedrazzi) [inline scripts/packaging/mac_distro.py:189]: Why did you add a default empty value for the directory?
If directory is empty, then this function does nothing and should probably not be called.

Comment (omichel) [reply scripts/packaging/mac_distro.py]: I used it before to create the 'Contents' folder, but I don't use it any more, so I just removed it.

**Comment by omichel** [comment]
Note: you can test that a binary file, e.g. `webots` contains both architectures with `lipo -archs webots`. It also works with `.dylib` dynamic libraries and `.a` static libraries.

**Review by ygoumaz — COMMENTED** [review]
The Matlab issue seems to be completely fixed. 

On the other side, I noticed a few additional wrong behaviors with the compilation on MacOS:

- `make clean` doesn't clean the lib/controller/matlab folder correctly.
- It is impossible to call `make release` or `make distrib` multiple time in a row, as some directories already exists.
- It is impossible to make a distribution. The process crashes at step "checking RPATH system" with the following errors:
Dependency error:
  - File: ./resources/lua/modules/gd/gd.dylib
  - Dependency: @rpath/lib/webots/liblua.dylib

  Dependency error:
  - File: ./resources/lua/modules/gd/gd.dylib
  - Dependency: @rpath/lib/webots/libgd.3.dylib

**Comment by omichel** [comment]
OK, I believe I fixed all the issues you reported.

**Comment by omichel** [comment]
🎉
